diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index eb69f3e..50d43ae 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -150,9 +150,9 @@ KBUILD_DTBS	:= dtbs
 
 KBUILD_DTBO_IMG := dtbo.img
 
-ifeq ($(CONFIG_BUILD_ARM64_DT_OVERLAY),y)
-export DTC_FLAGS := -@
-endif
+# ifeq ($(CONFIG_BUILD_ARM64_DT_OVERLAY),y)
+# export DTC_FLAGS := -@
+# endif
 
 all:	$(KBUILD_DTBS) $(KBUILD_TARGET)
 
@@ -182,9 +182,9 @@ Image-dtb: vmlinux scripts dtbs
 Image.gz-dtb: vmlinux scripts dtbs Image.gz
 	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
 
-ifeq ($(CONFIG_BUILD_ARM64_DT_OVERLAY),y)
-$(KBUILD_DTBO_IMG): dtbs
-	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
+# ifeq ($(CONFIG_BUILD_ARM64_DT_OVERLAY),y)
+# $(KBUILD_DTBO_IMG): dtbs
+# 	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
 
 all: $(KBUILD_DTBO_IMG)
 endif
diff --git a/arch/arm64/boot/Makefile b/arch/arm64/boot/Makefile
index 4e1fc8b..710ee18 100644
--- a/arch/arm64/boot/Makefile
+++ b/arch/arm64/boot/Makefile
@@ -18,7 +18,7 @@ include $(srctree)/arch/arm64/boot/dts/Makefile
 
 OBJCOPYFLAGS_Image :=-O binary -R .note -R .note.gnu.build-id -R .comment -S
 
-targets := Image Image.bz2 Image.gz Image.lz4 Image.lzma Image.lzo dtbo.img
+targets := Image Image.bz2 Image.gz Image.lz4 Image.lzma Image.lzo
 
 DTB_NAMES := $(subst $\",,$(CONFIG_BUILD_ARM64_APPENDED_DTB_IMAGE_NAMES))
 ifneq ($(DTB_NAMES),)
