diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index ef4dce6..5484ee2 100644
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -198,21 +198,6 @@ if test ! "$srctree" -ef .; then
 fi
 
 # CONFIG_LOCALVERSION and LOCALVERSION (if set)
-res="${res}${CONFIG_LOCALVERSION}${LOCALVERSION}"
-
-# scm version string if not at a tagged commit
-if test "$CONFIG_LOCALVERSION_AUTO" = "y"; then
-	# full scm version string
-	res="$res$(scm_version)"
-else
-	# append a plus sign if the repository is not in a clean
-	# annotated or signed tagged state (as git describe only
-	# looks at signed or annotated tags - git tag -a/-s) and
-	# LOCALVERSION= is not specified
-	if test "${LOCALVERSION+set}" != "set"; then
-		scm=$(scm_version --short)
-		res="$res${scm:++}"
-	fi
-fi
+res="${res}${CONFIG_LOCALVERSION}-g92c089fc2d37"
 
 echo "$res"
\ No newline at end of file
